static/js/message_events.js

```diff
@@ -139,7 +139,8 @@ export function insert_new_messages(messages, sent_by_this_client) {
             );
         }
     } else {
-        // we're in the home view, so update its list
+        // User is not narrowed to a stream / topic / PM, so we just update the home message
+        // list which is always kept maintained.
         render_info = message_util.add_new_messages(messages, message_lists.home);
     }
```

Comment:
Thinking more broadly about our goals here... we should possibly take this opportunity to rewrite this block of code as a loop over `all_rendered_message_lists`; we want to call `add_new_messages` on all of those, doing `maybe_add_narrowed_messages` if `!can_apply_locally`, with the `msg_list !== message_lists.current` check at the top of that function doing `narrow_state.is_message_feed_visible`...